thirdExample.js

```diff
@@ -0,0 +1,24 @@
+import React, {useState, useMemo} from 'react';
+
+const ThirdExample = ({someProp}) => {
+    if (someProp) {
+        //we can't put hooks in conditions because
+        //react doesn't know exact ordering of hooks
+        //order is important for React for how React associate hook calls with component
```

Comment:
One thing to add here is, that while React stores those hooks in an array, not only ordering is important, but also length of this array. Every time a component is rendered, every hook should be declared in this array in the same order. So no `if`s, loops and functions.
The component also shouldn't have any `early returns`.

Reply:
Good to know what exactly causes this limitation, thanks.